### PR TITLE
Paginate list_tables

### LINKED ISSUE #25: I am trying it with my clickhouse cloud instance and the project always crashes when list_tables is called on the databases with more then 100 tables.
I am trying it with my clickhouse cloud instance and the project always crashes when list_tables is called on the databases with more then 100 tables.

Anything I can do the fix it?

Pasting some log lines if they help. Let me know if you need more info here.
```log
 + Exception Group Traceback (most recent call last):
  |   File "/Users/puneetsutar/.cache/uv/archive-v0/UJ0ZpE2MZyeSn-kocDSZB/bin/mcp-clickhouse", line 12, in <module>
  |     sys.exit(main())
  |              ~~~~^^
  |   File "/Users/puneetsutar/.cache/uv/archive-v0/UJ0ZpE2MZyeSn-kocDSZB/lib/python3.13/site-packages/mcp_clickhouse/main.py", line 5, in main
  |     mcp.run()
  |     ~~~~~~~^^
  |   File "/Users/puneetsutar/.cache/uv/archive-v0/UJ0ZpE2MZyeSn-kocDSZB/lib/python3.13/site-packages/mcp/server/fastmcp/server.py", line 159, in run
  |     anyio.run(self.run_stdio_async)
  |     ~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^
  |   File "/Users/puneetsutar/.cache/uv/archive-v0/UJ0ZpE2MZyeSn-kocDSZB/lib/python3.13/site-packages/anyio/_core/_eventloop.py", line 74, in run
  |     return async_backend.run(func, args, {}, backend_options)
  |            ~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  |   File "/Users/puneetsutar/.cache/uv/archive-v0/UJ0ZpE2MZyeSn-kocDSZB/lib/python3.13/site-packages/anyio/_backends/_asyncio.py", line 2310, in run
  |     return runner.run(wrapper())
  |            ~~~~~~~~~~^^^^^^^^^^^
  |   File "/opt/homebrew/Cellar/python@3.13/3.13.2/Frameworks/Python.framework/Versions/3.13/lib/python3.13/asyncio/runners.py", line 118, in run
  |     return self._loop.run_until_complete(task)
  |            ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^
  |   File "/opt/homebrew/Cellar/python@3.13/3.13.2/Frameworks/Python.framework/Versions/3.13/lib/python3.13/asyncio/base_events.py", line 725, in run_until_complete
  |     return future.result()
  |            ~~~~~~~~~~~~~^^
  |   File "/Users/puneetsutar/.cache/uv/archive-v0/UJ0ZpE2MZyeSn-kocDSZB/lib/python3.13/site-packages/anyio/_backends/_asyncio.py", line 2298, in wrapper
  |     return await func(*args)
  |            ^^^^^^^^^^^^^^^^^
  |   File "/Users/puneetsutar/.cache/uv/archive-v0/UJ0ZpE2MZyeSn-kocDSZB/lib/python3.13/site-packages/mcp/server/fastmcp/server.py", line 460, in run_stdio_async
  |     async with stdio_server() as (read_stream, write_stream):
  |                ~~~~~~~~~~~~^^
  |   File "/opt/homebrew/Cellar/python@3.13/3.13.2/Frameworks/Python.framework/Versions/3.13/lib/python3.13/contextlib.py", line 221, in __aexit__
  |     await anext(self.gen)
  |   File "/Users/puneetsutar/.cache/uv/archive-v0/UJ0ZpE2MZyeSn-kocDSZB/lib/python3.13/site-packages/mcp/server/stdio.py", line 83, in stdio_server
  |     async with anyio.create_task_group() as tg:
  |                ~~~~~~~~~~~~~~~~~~~~~~~^^
  |   File "/Users/puneetsutar/.cache/uv/archive-v0/UJ0ZpE2MZyeSn-kocDSZB/lib/python3.13/site-packages/anyio/_backends/_asyncio.py", line 772, in __aexit__
  |     raise BaseExceptionGroup(
  |         "unhandled errors in a TaskGroup", self._exceptions
  |     ) from None
  | ExceptionGroup: unhandled errors in a TaskGroup (1 sub-exception)
  +-+---------------- 1 ----------------
    | Traceback (most recent call last):
    |   File "/Users/puneetsutar/.cache/uv/archive-v0/UJ0ZpE2MZyeSn-kocDSZB/lib/python3.13/site-packages/mcp/server/stdio.py", line 69, in stdin_reader
    |     await read_stream_writer.send(message)
    |   File "/Users/puneetsutar/.cache/uv/archive-v0/UJ0ZpE2MZyeSn-kocDSZB/lib/python3.13/site-packages/anyio/streams/memory.py", line 255, in send
    |     raise BrokenResourceError from None
    | anyio.BrokenResourceError
    +------------------------------------
2025-03-30T21:23:43.811Z [mcp-clickhouse] [info] Server transport closed
2025-03-30T21:23:43.811Z [mcp-clickhouse] [info] Client transport closed
2025-03-30T21:23:43.811Z [mcp-clickhouse] [info] Server transport closed unexpectedly, this is likely due to the process exiting early. If you are developing this MCP server you can add output to stderr (i.e. `console.error('...')` in JavaScript, `print('...', file=sys.stderr)` in python) and it will appear in this log.
2025-03-30T21:23:43.811Z [mcp-clickhouse] [error] Server disconnected. For troubleshooting guidance, please visit our [debugging documentation](https://modelcontextprotocol.io/docs/tools/debugging) {"context":"connection"}
2025-03-30T21:23:43.812Z [mcp-clickhouse] [info] Client transport closed
```

### PR DESCRIPTION
**Pull request by joe-clickhouse**
# Summary
Most of the original work for this was done by @iskakaushik in #34. However, in trying to resolve several month's worth of conflict on main I accidentally got it into a funky state (sorry!). So, I decided to rebuild pagination for `list_tables` on top of current main rather than merging main. Note that I also added a metadata trimming switch `include_detailed_columns` to reduce payload size when needed.

# Key Changes
- Return type: `list_tables` now returns a `dict` with the keys `tables`, `next_page_token`, and `total_tables`. It was a list previously
- Pagination: `page_size` (default 50) and `page_token` (one‑time tokens, 1‑hour TTL via cache)
- Metadata trimming: `include_detailed_columns` (defaults to `True`) to preserve previous behavior but can be set to `False` to omit column metadata which will create lighter responses while preserving the full `create_table_query`
- Token safety: mismatched database/filters/metadata invalidates the token and resets to first page
- Dependency: add `cachetools` for TTL cache
- New: `tests/test_pagination.py`
- Updates: `tests/test_tool.py`, `tests/test_mcp_server.py` has been updated to reflect the new response shape and edge cases

# LLM Usage
- Users can request a “lightweight listing” or “omit column details” and the assistant will call `list_tables` with `include_detailed_columns=false` and/or a smaller `page_size`.
- When `next_page_token` is returned, the assistant paginates by passing it back until the user narrows scope or coverage is sufficient.
 
-----------------------

Closes #25

### CODE REVIEW

File: mcp_clickhouse/mcp_server.py [hunk 1]
```diff
@@ -135,36 +137,252 @@ def list_databases():
...
+
+
+def get_paginated_table_data(
+    client,
+    database: str,
+    table_names: List[str],
+    start_idx: int,
+    page_size: int,
+    include_detailed_columns: bool = True,
+) -> tuple[List[Table], int, bool]:
```
[the remark this answers is not in the document]
Comment (joe-clickhouse) [reply mcp_clickhouse/mcp_server.py:181]: We require Python>=3.10. Won't fix.

File: mcp_clickhouse/mcp_server.py [hunk 2]
```diff
@@ -135,36 +137,252 @@ def list_databases():
...
+    """Get detailed information for a page of tables.
+
+    Args:
+        client: ClickHouse client
+        database: Database name
+        table_names: List of all table names to paginate
+        start_idx: Starting index for pagination
+        page_size: Number of tables per page
+        like: Optional LIKE filter
+        not_like: Optional NOT LIKE filter
```
[the remark this answers is not in the document]
Comment (joe-clickhouse) [reply mcp_clickhouse/mcp_server.py]: Yes, this is a fair point. I removed the unused args from sig but not the docstring. Will fix.

**Review by iskakaushik — APPROVED** [review]
LGTM
